models/models.js

```diff
@@ -16,12 +16,41 @@ exports.fetchTopics = () => {
   return db.query('SELECT * FROM topics;').then(({ rows }) => rows);
 };
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      'SELECT articles.*, COUNT(articles.article_id) as comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;'
-    )
-    .then(({ rows }) => rows);
+exports.fetchArticles = (sortBy = 'created_at', order = 'DESC', topic) => {
+  const allowedValues = {
+    sortBy: [
+      'article_id',
+      'title',
+      'topic',
+      'author',
+      'body',
+      'created_at',
+      'votes',
+      'comment_count',
+    ],
+    order: ['ASC', 'DESC'],
+  };
+
+  if (
+    !allowedValues.sortBy.includes(sortBy) ||
+    !allowedValues.order.includes(order)
+  ) {
+    return Promise.reject({ status: 400, msg: 'Bad Request' });
+  }
```

Comment:
Great use of `||` here - keeping things DRY!
Suggested something similar for your error handler